### PR TITLE
Support Iter8 v0.2

### PR DESCRIPTION
**Pull request by jadeyliu939**
** Describe the change **
1. Iter8 v0.2 supports deployment/service-based experiment
2. add Host/Gateway support
3. Warning message if not running Iter8 v0.2

** Issue reference **
[#2007](https://github.com/kiali/kiali/issues/2007)

** Backwards compatible? **
This PR depends on [PR#3033](https://github.com/kiali/kiali/pull/3033)

[ ] Is your pull-request introducing changes in behaviour?

_Describe the changes if appropriate. E.g. a new link, a change on what a click does, etc_

** Screenshot **
1. Iter8 v0.2 supports deployment/service-based experiment
A. Deployment-Based
Screen Shot 2020-07-02 at 8 31 05 PM
B. Service-Based
Screen Shot 2020-07-02 at 8 31 22 PM
C. Experiment Listing page show the two different experiment ( deployment-base vs service-based)
Screen Shot 2020-07-02 at 8 32 13 PM

2. add Host/Gateway support
Screen Shot 2020-07-02 at 8 31 31 PM

3. Warning message if not running Iter8 v0.2
Screen Shot 2020-07-02 at 8 37 50 PM

### CODE REVIEW

File: src/pages/extensions/iter8/Iter8ExperimentDetails/ExperimentCreatePage.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.pf-c-switch__toggle {
```
Comment (lucasponce) [inline src/pages/extensions/iter8/Iter8ExperimentDetails/ExperimentCreatePage.scss:1]: I'm afraid this breaks the style in other pages:
image

So, I'd suggest to skip the change of color.
I know we have a pending review of the design of this form later, perhaps it's something we can adjust with the other Istio forms.

**Comment by lucasponce** [comment]
I'm fine to resolve the comment with the colors and progress but I wonder if we may want to use some consistency with conditional rendering used in the 'Create New Istio Config' forms.

So, things that I could be cheap to change in a next iteration (not on this PR, of course):

- Deployment/Service can be a dropdown with two values similar to this and showing rendering displaying:

image

image

- "Traffic Control" and "Gateways" section can also use a similar pattern like this (instead to put an "optional" in the title):

image

image

I mean in that way we can try to consolidate these forms, under the "Create New Istio Config" there are several examples that perhaps can help to the Iter8 form as well.

**Comment by lucasponce** [comment]
Also, another suggestion is if we can use a similar layout of "single column (label + component)" to be consitent others forms:

image

At least in the first group of common fields in iter8 form:
image

Which is using "two columns of label+component pairs".

**Comment by jadeyliu939** [comment]
@lucasponce  I have discovered a problem with both namespaces selections ( the default one, and the one I have) in the create page. I notice in the 3scale create page, you have your namespace input disabled but use the default one, but in iter8, i can not have multiple namespaces in creating experiment. So I update defaultSecondaryMashead.tsx and namespaceDropdown.tsx. In defaultSecondaryMashead.tsx, I make isDisabled:true for iter8 create. 

`<NamespaceDropdownContainer>`  already has a prop:disabled, but the prop is not being used currently,  So I added `isDisabled={this.props.disabled}` for `<DropdownToggle>`. Let me know if the change is ok.

**Comment by jadeyliu939** [comment]
Screen Shot 2020-07-29 at 4 10 42 PM
Screen Shot 2020-07-29 at 4 10 25 PM

**Comment by lucasponce** [comment]
@jadeyliu939 I'd ask you if you can perhaps re-arrange this work (in this or other clean PR) to just send the commits needed for the Iter8 work.

From my review I see a very large number of commits to be merge in master (52):

image

And that's due not rebasing the previous PR and introduce some diverge from master.

I'm +1 with the work, and I'd like to consolidate it and refine later in master, but we can't merge this as it is (it may introduce problems with other work easily).

Thank you very much !
Lucas

**Comment by jadeyliu939** [comment]
CLosed, New PR open [PR#1868](https://github.com/kiali/kiali-ui/pull/1868)

**Pull request closed without merging**